<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>1. Consider the data given below. Can we claim that the average number of fruits on a tree is 70? 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider number of fruits on a tree:

            number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

In [1]:
number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

The Hypothesis is:-

Ho:The mean Number of fruits on a tree is eqaul to 70,i.e µ=70

Against

Ha:The mean Number of fruits on a tree is not eqaul to 70 70,i.e µ≠70 

In [2]:
from scipy.stats import ttest_1samp

ttest_1samp(number_fruits,70)

Ttest_1sampResult(statistic=-1.36854291090804, pvalue=0.19270344894968697)

p-value is greater than 0.05,hence we fail to reject Ho and the claim is right.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>2. The courier company has two offices in NYC. They are interested to know which office is busier. The number of parcels received for 15 days is given below. Can we claim that both offices are equally busy?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider the number of parcels in office A:
            office_A = [26, 56, 48, 56, 79, 67, 35, 46, 87, 53, 67, 89, 65, 42, 34]

            Consider the number of parcels in office B:
            office_B = [34, 45, 87, 57, 45, 42, 34, 45, 34, 67, 89, 43, 87, 95, 95]

In [7]:
import pandas as pd 
office_A = pd.Series([26, 56, 48, 56, 79, 67, 35, 46, 87, 53, 67, 89, 65, 42, 34])
office_B = pd.Series([34, 45, 87, 57, 45, 42, 34, 45, 34, 67, 89, 43, 87, 95, 95])

The Hypothesis is:-

Ho:The average A is eqaul to average  B,i.e µA=µB

Against

Ha:The average A is not eqaul to average B,i.e µA≠µB

In [8]:
from scipy.stats import ttest_ind

ttest_ind(office_A,office_B)

Ttest_indResult(statistic=-0.41183741857533557, pvalue=0.6835943945208511)

p-value is greater than 0.05,hence we fail to reject Ho and the claim is right

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>3. To improve the attitude of officers, they had to complete a training program. Can we claim that the training program was effective? 
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

The data of 10 officers scores is given below:

<table>
    <tr>
        <th> Officer </th>
        <th> Attitude Before </th>
        <th> Attitude After </th>
    </tr>
     <tr>
        <td> 1 </td>
        <td> 3 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 2 </td>
        <td> 5 </td>
        <td> 5 </td>
    </tr>
    <tr>
        <td> 3 </td>
        <td> 2 </td>
        <td> 3 </td>
    </tr>
    <tr>
        <td> 4 </td>
        <td> 2 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 5 </td>
        <td> 4 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 6 </td>
        <td> 4 </td>
        <td> 4 </td>
    </tr>
    <tr>
        <td> 7 </td>
        <td> 3 </td>
        <td> 3 </td>
    </tr>
    <tr>
        <td> 8 </td>
        <td> 5 </td>
        <td> 2 </td>
    </tr>
    <tr>
        <td> 9 </td>
        <td> 3 </td>
        <td> 2 </td>
    </tr>
    <tr>
        <td> 10 </td>
        <td> 4 </td>
        <td> 5 </td>
    </tr>
</table>

In [13]:
import os
os.chdir(r"D:\Imarticus\Take Home Practice\Statistics Take-Home Practice")
df=pd.read_excel("Statistics_data.xlsx",sheet_name='Sheet1')
df

Officer  Attitude Before  Attitude After
0        1                3               4
1        2                5               5
2        3                2               3
3        4                2               4
4        5                4               4
5        6                4               4
6        7                3               3
7        8                5               2
8        9                3               2
9       10                4               5

The Hypothesis is:-

Ho:The average Attitude Before is eqaul to average  Attitude After,i.e µA=µB

Against

Ha:The average Attitude Before is not eqaul to average Attitude After,i.e µA≠µB

In [15]:
from scipy.stats import ttest_ind

ttest_ind(df["Attitude Before"],df["Attitude After"])

Ttest_indResult(statistic=-0.2075143391598226, pvalue=0.8379378556799146)

p-value is greater than 0.05,hence we fail to reject Ho .

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>4. During an economic crisis, a researcher thought that the salaries of IT professionals are affected. Test whether the average monthly salary has gone below 65000 dollars.
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Salaries of corporate professionals in thousand dollars:

            salary = [56, 48, 56, 79, 67, 55, 46, 87, 53, 67, 89, 65, 62, 45, 87, 57, 45, 
                      42, 45, 67, 89, 43, 87, 95, 92, 56, 67, 89, 54, 62, 78, 43, 76, 65,
                      56, 58, 98, 56, 78, 79, 45, 34, 35, 46, 47, 67, 37, 45, 51, 65, 65]

In [16]:
salary = [56, 48, 56, 79, 67, 55, 46, 87, 53, 67, 89, 65, 62, 45, 87, 57, 45, 42, 45, 67, 89, 43, 87, 95, 92, 56, 67, 89, 54, 62, 78, 43, 76, 65,56, 58, 98, 56, 78, 79, 45, 34, 35, 46, 47, 67, 37, 45, 51, 65, 65]

The hypothesis is                                                                       
H<sub>0</sub>: The average salary of a corporate professional is equal to 65000, i.e $\mu$= 65000 <br>
AGAINST
H<sub>a</sub>: The average salary of a corporate professional is less than 65000, , i.e $\mu$ ≤ 65000

In [18]:
from statsmodels.stats import weightstats as stests

stests.ztest(salary, value = 65, alternative = "smaller")

(-1.1250934292594312, 0.1302747226990263)

P-value (0.13) is more than 0.05 , hence we fail to reject H0.

<table align="left">
    <tr>
        <td width="6%">
            <img src="scala_q.png">
        </td>
        <td>
            <div align="left", style="font-size:120%">
                <font color="#21618C">
                    <b>5. Consider the data given below. Can we conclude that the data is normally distributed?
</b>
                </font>
            </div>
        </td>
    </tr>
</table>

            Consider number of fruits on a tree:
            number_fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

As, the data is less than 5000 ,hence we can use shapiro method to test the normality

In [19]:
fruits = [64, 56, 48, 56, 79, 67, 65, 86, 87, 53, 67, 89, 65, 42, 46]

H<sub>0</sub>: The data is normally distributed<br>
**AGAINST**<br>
H<sub>a</sub>: The data is not normally distributed

In [21]:
from scipy.stats import shapiro

stat, p = shapiro(fruits)

print('The test statistic is', stat, 'and its p-value is', p)

The test statistic is 0.9354038238525391 and its p-value is 0.3280726969242096


p-value (0.328) is more than 0.05 (level of significance), hence we fail to reject H0.Hence, the data is normally distributed.